src/Service/Services/ResolverTypeInterceptor.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.DataApiBuilder.Service.Services;
+using HotChocolate.Configuration;
+using HotChocolate.Resolvers;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors.Definitions;
+
+internal sealed class ResolverTypeInterceptor : TypeInterceptor
+{
+    private readonly FieldMiddlewareDefinition _queryMiddleware;
+    private readonly FieldMiddlewareDefinition _mutationMiddleware;
+    private readonly PureFieldDelegate _leafFieldResolver;
+    private readonly PureFieldDelegate _objectFieldResolver;
+    private readonly PureFieldDelegate _listFieldResolver;
+    
+    public ResolverTypeInterceptor(ExecutionHelper executionHelper)
+    {
+        _queryMiddleware = 
+            new FieldMiddlewareDefinition(
+                next => async context =>
+                {
+                    await executionHelper.ExecuteQueryAsync(context).ConfigureAwait(false);
+                    await next(context).ConfigureAwait(false);
+                });
+        
+        _mutationMiddleware = 
+            new FieldMiddlewareDefinition(
+                next => async context =>
+                {
+                    await executionHelper.ExecuteMutateAsync(context).ConfigureAwait(false);
+                    await next(context).ConfigureAwait(false);
+                });
+
+        _leafFieldResolver = ctx => ExecutionHelper.ExecuteLeafField(ctx);
+        _objectFieldResolver = ctx => executionHelper.ExecuteObjectField(ctx);
```

Comment:
Given that these resolvers: `ExecuteObjectField/ExecuteListField` dont return `JsonDocument` whereas the mutation and query resolvers do, who are the callers of these resolvers that are able to consume `JsonElement`

Reply:
makes sense, thanks for the explanation!